stablehlo/tests/interpret_if.mlir

```diff
@@ -0,0 +1,27 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
+
+// CHECK-LABEL: Evaluated results of function: if
+func.func @if() -> (tensor<2xi64>, tensor<2xi64>) {
+  %pred = stablehlo.constant dense<false> : tensor<i1>
+  %result0, %result1 = "stablehlo.if"(%pred) ({
+    %0 = stablehlo.constant dense<0> : tensor<2xi64>
+    "stablehlo.return"(%0, %0) : (tensor<2xi64>, tensor<2xi64>) -> ()
+  }, {
+    %false_branch_pred = stablehlo.constant dense<true> : tensor<i1>
+    %false_branch_result = "stablehlo.if"(%false_branch_pred) ({
+      %1 = stablehlo.constant dense<1> : tensor<2xi64>
+      "stablehlo.return"(%1) : (tensor<2xi64>) -> ()
+    }, {
+      %2 = stablehlo.constant dense<2> : tensor<2xi64>
+      "stablehlo.return"(%2) : (tensor<2xi64>) -> ()
+    }) : (tensor<i1>) -> tensor<2xi64>
+    "stablehlo.return"(%false_branch_result, %false_branch_result) : (tensor<2xi64>, tensor<2xi64>) -> ()
+  }) : (tensor<i1>) -> (tensor<2xi64>, tensor<2xi64>)
+  func.return %result0, %result1 : tensor<2xi64>, tensor<2xi64>
```

Comment:
Can we perhaps have two tests - one for `if_true` and another one for `if_false`? That'll be easier to read than a nested test.

Reply:
I intentionally included this one to comprehensively test that:
1) Both if_true and if_false works
2) Nested if works
3) Can return variadic outputs

I agree we should have two tests since the code has two logical paths as per the testing guideline. However, we would still need to test 2) and 3), which we need to include in both paths to have minimal tests (unless such positive tests are in ops_stablehlo.mlir, but still debatable). Thoughts?

Reply:
Agreed by 1 and 3, but why do we need to test 2? If has two regions, but these regions aren't special, right?

Reply:
Agreed with not testing nested behaviors. This behavior should be tested separately from the ops.